feat: Vuetify3 support

mariobuikhuizen opened the pull request:
Needs https://github.com/widgetti/ipyvue/pull/82

TODO:
 - Theme support
 - check and update documentation and examples
 - make nodeps version work

mariobuikhuizen committed:
feat: Vuetify3 support

mariobuikhuizen committed:
feat: Vuetify3 support

mariobuikhuizen committed:
build: sort components and properties in the generated code

Now that we store generated code in git, having a stable sort order
will allow us to see API changes easily in the diff.

mariobuikhuizen committed:
build: sort components and properties in the generated code

Now that we store generated code in git, having a stable sort order
will allow us to see API changes easily in the diff.

mariobuikhuizen committed:
build: use latest node LTS version

mariobuikhuizen committed:
build: use latest node LTS version

mariobuikhuizen committed:
feat: keep same module structure for generated python code

mariobuikhuizen committed:
feat: keep same module structure for generated python code

mariobuikhuizen committed:
chore: bump version 1.8.10 -> 3.0.0.dev0

mariobuikhuizen committed:
chore: bump version 1.8.10 -> 3.0.0.dev0

maartenbreddels committed:
chore: require ipyvue>=3.0.0.dev0

maartenbreddels committed:
chore: require ipyvue>=3.0.0.dev0

maartenbreddels committed:
chore: update reacton component

maartenbreddels committed:
chore: update reacton component

maartenbreddels committed:
chore: format code with new line width

maartenbreddels committed:
chore: format code with new line width

maartenbreddels committed:
fix: typo pm->npm

maartenbreddels committed:
fix: typo pm->npm

maartenbreddels committed:
temp: install solara from branch

maartenbreddels committed:
temp: install solara from branch

mariobuikhuizen committed:
docs: make example notebooks work

mariobuikhuizen committed:
docs: make example notebooks work

maartenbreddels committed:
temp: install ipyvue from branch

maartenbreddels committed:
temp: install ipyvue from branch

maartenbreddels committed:
fix: no need for special nodeps view

maartenbreddels committed:
fix: no need for special nodeps view

mariobuikhuizen committed:
feat: reintroduce vueRender so we can keep the templates the same

mariobuikhuizen committed:
feat: reintroduce vueRender so we can keep the templates the same

mariobuikhuizen committed:
chore: use new prepend-icon prop of vuetify3 in ui-tests

mariobuikhuizen committed:
chore: use new prepend-icon prop of vuetify3 in ui-tests

mariobuikhuizen committed:
build: use correct cli param in ui-tests

mariobuikhuizen committed:
build: use correct cli param in ui-tests

mariobuikhuizen committed:
build: accept the small change in rendering for ui-tests

mariobuikhuizen committed:
build: accept the small change in rendering for ui-tests

mariobuikhuizen committed:
build: accept visual change in ui-test

mariobuikhuizen committed:
build: accept visual change in ui-test

mariobuikhuizen committed:
feat: add theme support

mariobuikhuizen committed:
feat: add theme support

mariobuikhuizen committed:
chore: remove commented-out code, update js model versions

mariobuikhuizen committed:
chore: remove commented-out code, update js model versions

mariobuikhuizen committed:
chore: release v3.0.0.alpha1

mariobuikhuizen committed:
chore: release v3.0.0.alpha1

mariobuikhuizen committed:
build: use correct version in pypackage.toml and add dry-run

mariobuikhuizen committed:
build: use correct version in pypackage.toml and add dry-run

mariobuikhuizen committed:
fix: relax ipyvue version constraint a bit

mariobuikhuizen committed:
fix: relax ipyvue version constraint a bit

mariobuikhuizen committed:
chore: bump ipyvue version

mariobuikhuizen committed:
chore: bump ipyvue version

mariobuikhuizen committed:
chore: release v3.0.0.alpha2

mariobuikhuizen committed:
chore: release v3.0.0.alpha2

iisakkirotko committed:
chore: add build all script

iisakkirotko committed:
chore: add build all script

iisakkirotko committed:
feat: scope vuetify styles with :where

Ports the changes made in https://github.com/widgetti/ipyvuetify/pull/327 to version 3 alpha

iisakkirotko committed:
feat: scope vuetify styles with :where

Ports the changes made in https://github.com/widgetti/ipyvuetify/pull/327 to version 3 alpha

iisakkirotko committed:
feat: export Vue and Vuetify

iisakkirotko committed:
feat: export Vue and Vuetify

iisakkirotko committed:
chore: bump external actions

iisakkirotko committed:
chore: bump external actions

iisakkirotko committed:
chore: use jupyter-vue version 3

iisakkirotko committed:
chore: use jupyter-vue version 3

iisakkirotko committed:
chore: pin pytest-playwright

See https://github.com/widgetti/solara/issues/913

iisakkirotko committed:
chore: move test runners to ubuntu-22.04

Ubuntu 20.04 runners will be disabled on 1.4.2025, see https://github.com/actions/runner-images/issues/11101

iisakkirotko committed:
chore: move test runners to ubuntu-22.04

Ubuntu 20.04 runners will be disabled on 1.4.2025, see https://github.com/actions/runner-images/issues/11101

iisakkirotko committed:
chore: update officially supported Python versions to 3.6-3.13

iisakkirotko committed:
chore: update officially supported Python versions to 3.6-3.13

iisakkirotko committed:
fix: revert css extraction (#332)

https://github.com/widgetti/ipyvuetify/pull/327 broke embedded versions where the extractes CSS wasn't explicitly imported.

iisakkirotko committed:
fix: revert css extraction (#332)

https://github.com/widgetti/ipyvuetify/pull/327 broke embedded versions where the extractes CSS wasn't explicitly imported.

iisakkirotko committed:
fix: ensure correct styles are excluded from scoping

For some reason the styles.css file wasn't excluded from scoping. This should ensure it is.

iisakkirotko committed:
fix: ensure correct styles are excluded from scoping

For some reason the styles.css file wasn't excluded from scoping. This should ensure it is.

mariobuikhuizen committed:
chore: release v3.0.0.alpha3

mariobuikhuizen committed:
chore: release v3.0.0.alpha3

12rambau commented:
Hi ipyvuetify team, just to let you know that vuetify3 has made a very good job at tidying the way they built the API documetnation I would happilly go back to my documentation PR once this one is done as it will be much easier to recover the text associated to each parameter directly from their repository. 

If the improvement of the documentation is not a wanted feature anymore, let me know I'll drop the PR I have on stale on my fork.

mariobuikhuizen committed:
fix: get the right vue export for nodeps for solara

mariobuikhuizen committed:
fix: get the right vue export for nodeps for solara

mariobuikhuizen committed:
publish dev release to github releases

mariobuikhuizen committed:
publish dev release to github releases

mariobuikhuizen committed:
add new arguments for reacton to VuetifyTemplate

mariobuikhuizen committed:
fix github release download

mariobuikhuizen committed:
fix github release download

mariobuikhuizen committed:
use object store instead of github releases

mariobuikhuizen committed:
use object store instead of github releases

mariobuikhuizen committed:
expose vutify namespace to templates

mariobuikhuizen committed:
expose vutify namespace to templates

mariobuikhuizen committed:
fix dependecy issue with solara-test

mariobuikhuizen committed:
fix dependecy issue with solara-test

mariobuikhuizen committed:
Don't create a new theme for each view

mariobuikhuizen committed:
Don't create a new theme for each view

mariobuikhuizen committed:
chore: update package-lock after rebase
pick 820559517 # fix: main styles were not applied becaus of invalid comment

mariobuikhuizen committed:
chore: update docs

mariobuikhuizen committed:
fix: make datatable available for solara

mariobuikhuizen committed:
ci: actually publish dev build on PR push

mariobuikhuizen committed:
refactor: Override Vue component factory for Vuetify views

Move the Vuetify-specific render setup into vueComponent(), overriding the
base VueView implementation. This gives Solara and other embedders a stable
component definition while preserving the normal Vuetify plugin/theme setup.

mariobuikhuizen committed:
feat: add support for custom theme colors

mariobuikhuizen committed:
fix: export all lab components

mariobuikhuizen committed:
ci: use npm Trusted Publishing

mariobuikhuizen committed:
chore: release v3.0.0.alpha4

mariobuikhuizen committed:
chore: update Vuetify to 3.12.9

mariobuikhuizen committed:
chore: release v3.0.0.alpha5

mariobuikhuizen committed:
chore: temporarily disable solara ui-tests

mariobuikhuizen committed:
fix: theme from solara app is not applied

Use useTheme() instead of relying on internal implementaion
details.

mariobuikhuizen committed:
chore: release v3.0.0.alpha6

mariobuikhuizen committed:
chore: update released dependencies

mariobuikhuizen committed:
CI: fix jupyter lab 4.6 plugin build

mariobuikhuizen committed:
CI: update github actions dependencies

mariobuikhuizen committed:
build: update JS dependencies to address security advisories

mariobuikhuizen committed:
chore: update readme to Vuetify 3
